05.ls/ls_1.rb

```diff
@@ -1,32 +1,56 @@
 # frozen_string_literal: true
 
-require 'optparse'
+require 'etc'
 
-COLUMN_COUNT = 3
+FILE_TYPE = {
+  'file' => '-',
+  'link' => 'l',
+  'directory' => 'd'
+}.freeze
+
+PERMISSION_TYPE = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
 
 def main
-  options = ARGV.getopts('r')
-  files = options['r'] ? Dir.glob('*').reverse : Dir.glob('*')
-  transposed_files = build_transposed_files(files)
-  max_file_size = files.map(&:size).max
-  output_files(transposed_files, max_file_size)
+  files = Dir.glob('*')
+  output_total_block_count(files)
+  output_files(files)
+end
+
+def output_total_block_count(files)
+  total_block_count = files.sum { |file_count| File.lstat(file_count).blocks }
+  puts "total #{total_block_count}"
 end
 
-def build_transposed_files(files)
-  row_count = (files.size.to_f / COLUMN_COUNT).ceil
-  nested_files = files.each_slice(row_count).to_a
-  nested_files.each do |file_names|
-    file_names << nil while file_names.size < row_count
+def output_permission(stat_file)
+  permission_count = stat_file.mode.to_s(8).slice(-3, 3)
+  permission_type = permission_count.split('').map do |file_count|
+    PERMISSION_TYPE[file_count]
   end
-  nested_files.transpose
+  permission_type.join('').to_s.ljust(8)
```

Comment:
```suggestion
  permission_type.join.ljust(8)
```

joinの引数はなくても同じ結果になります。
あと、joinの戻り値はstringなのでto_sは不要です。

ちなみに`ljust(8)`は何のために付けていますか？（これを無くすとどういう問題が起きますか？）

Reply:
パーミッションの値8桁を左揃えにしようと考え`ljust(8)`をつけていましたが外しても同じ結果でした！ので無くしました。